test/kubernetes_unittest.cc

```diff
@@ -1951,4 +1966,52 @@ TEST_F(KubernetesTestFakeServerThreeWatchRetriesNodeLevelMetadata,
   EXPECT_TRUE(server->TerminateAllStreams(time::seconds(3)));
 }
 
+namespace {
+class MockSleeper : public Sleeper {
+ public:
+  MOCK_METHOD1(SleepFor, void(double seconds));
+};
+
+// Polls health_checker until it has at least one unhealthy component.
+// Returns false if newer timestamp not found after 3 seconds (polling
+// every 100 millis).
+bool WaitForUnhealthyComponents(const HealthChecker& health_checker) {
+  for (int i = 0; i < 30; i++){
+    if (health_checker.UnhealthyComponents().size() > 0) {
+      return true;
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+  return false;
+}
+}
+
+TEST_F(KubernetesTestFakeServerMaxThreeFailuresNodeLevelMetadata,
+       KubernetesUpdaterBackoffOnFailures) {
+  // Create a fake server representing the Kubernetes master, but
+  // don't register streams for the nodes or pods watch paths, so they
+  // will return 404s.
+  //
+  // This will cause the updater to backoff 3 times and then give up.
+  server->SetResponse("/api/v1/nodes?limit=1", "{}");
```

Comment:
Minor nit: this is a constant source of confusion for me. WDYT about moving it to a helper function such as `InitHealthyServer(server)` to reduce duplication and make the intent clear for the reader?

Reply:
Done.